### ISSUE TITLE
An array of nested struct not supported

### ISSUE THREAD

**Issue by pash7ka** [ethereum/solidity#3546]
If my contract has struct A, struct B where one field is A, and an array of B, it can not be compiled with an error ``UnimplementedFeatureError: Only in-memory reference type can be stored.``
```
contract ArrayOfNestedStructTest {
    struct A {
        uint a;
    }
    struct B {
        A b;        
    }
    B[] public bArr;
}
```
solc version: up to 0.4.21-nightly.2018.2.16+commit.3f7e82d0.Emscripten.clang
Expected: this contract should be compiled without errors and warnings.

**Comment by chriseth** [comment]
This used to generate an error `"Internal type is not allowed for public state variables."`, but now it is not clear how the signature of that getter would look like. Do we actually want to return all struct members? What if some of them are arrays?

**Comment by wbt** [comment]
The workaround for this is to remove the keyword `public` from 

    B[] public bArr;

and if you want to access something within bArr, write your own view function to do that.   
The error message could be much more helpful as a guide toward this strategy, though.

**Comment by chriseth** [comment]
This has been fixed by #3549
